### DISCUSSION TITLE
Play Asset Delivery ANE needs requestCellularDataConfirmation capability

### DISCUSSION THREAD

**Discussion by jimmymjing** [airsdk/Adobe-Runtime-Support#1581] [General]
Hello, is there an approximate release date for the AIR Package Manager tool? I wish to be able to use ` requestCellularDataConfirmation()` or `showCellularDataConfirmation() method for large Asset Pack downloads. Thanks!!

> If the download is larger than 150 MB and the user is not on Wi-Fi, the download does not start until the user explicitly gives their consent to proceed with the download using a mobile data connection. Similarly, if the download is large and the user loses Wi-Fi, the download is paused and explicit consent is required to proceed using a mobile data connection. A paused pack has state WAITING_FOR_WIFI. To trigger the UI flow to prompt the user for consent, use the requestCellularDataConfirmation() or showCellularDataConfirmation() method.

**Comment by ajwfrost** [comment]
I think this isn't specific to the APM tool -> instead it's a new feature we need to add to the PlayAssetDelivery ANE. Let me just double-check with the guy who had been working on it...

**Reply by marchbold** [reply]
Yeah I don't think this is related to the apm tool at all, are you asking about updating the specific package with the latest version of the ANE though? I can do that whenever the ANE is updated.

**Reply by jimmymjing** [reply]
I'm sorry that I made a mistake... I didn't mean APM at all. I thought we are going to have another ANE for the AssetPackManager(`com.google.android.play.core.assetpacks.AssetPackManager`).

**Reply by ajwfrost** [reply]
Yes so that Android `AssetPackManager` class is what we wrap up within the ANE via the `com.harman.extension.PlayAssetDelivery` class.
Will come back with an update on this particular issue later today...

thanks

**Reply by ajwfrost** [reply]
Can I just double-check, that you do get a status event sent out (`PlayAssetDeliveryEvent.PLAY_ASSET_UPDATE`) with the `WAITING_FOR_WIFI` value, when this happens?

**Reply by jimmymjing** [reply]
I haven't tested this yet. I can try it in 1 hour.

**Reply by jimmymjing** [reply]
@ajwfrost hi, I confirm I received the event wait for wifi.

**Comment by ajwfrost** [comment]
Thanks @jimmymjing  - we've just added a new AS3 method to `com.harman.extension.PlayAssetDelivery`:
```
public function showCellularDataConfirmation(assetPack:String):Boolean
```
You can call this when you get the event with the `WAITING_FOR_WIFI` value. It will return false if it doesn't know about the asset pack status or the status isn't at `WAITING_FOR_WIFI`, otherwise true to say that it's kicked off the request with the Android PlayAssetManager handler..

Please try using the ANE package from the below zip file..

thanks

[PlayAssetDelivery.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/7904929/PlayAssetDelivery.zip)

**Reply by jimmymjing** [reply]
It works like a charm! Thanks so much!!!

**Reply by ajwfrost** [reply]
Thanks - @marchbold , sorry, are you able to do another update into the repository so this latest version is picked up by APM? Thanks!

**Reply by marchbold** [reply]
v1.1.0 has been published with the above version of the extension.

**Reply by ajwfrost** [reply]
Thank you!
